File: 5.implementation/NT/redsun/src/main/webapp/static/js/services/DeliveryModesService.js


/**
 * Controller for ${entity.name}
 **/


app.factory('deliveryModeService', function($http, $rootScope) {
	return {
	
		// Insert.
		add: function(deliveryMode) {
			var urlServer = $rootScope.makePostURL("/deliverymode/insert");
			return $http.post(urlServer, deliveryMode);
		}
		
		// Update.
		, update: function(deliveryModeId, deliveryMode) {
			var urlServer = $rootScope.makePostURL("/deliverymode/update/" + deliveryModeId);
			return $http.put(urlServer, deliveryMode);
		}
		
		// Delete.
		, delete: function(deliveryModeId) {
			var urlServer = $rootScope.makePostURL("/deliverymode/delete/" + deliveryModeId);
			return $http.delete(urlServer);
		}
		
		// Get by Id.
		, getById: function(deliveryModeId) {
			var url = $rootScope.makeGetURL("/deliverymode/getbyid/" + deliveryModeId);
			return $http.get(url);
		}
		
		// List for page and filter.
		, getsForPageAndFilter: function(itemsPerPage, pageNo, deliveryMode) {
			var url = $rootScope.makeGetURL("/deliverymode/list/page/filter/" + itemsPerPage + "/" + pageNo);
			return $http.get(url, { params: deliveryMode });
		}
		
	}// return.
});

File: 5.implementation/NT/redsun-service/src/main/java/com/redsun/service/service/DeliveryModesService.java
package com.redsun.service.service;

import com.redsun.service.entities.DeliveryModes;
import com.redsun.service.entities.Result;

/**
 * DeliveryModes Service interface
 */
public interface DeliveryModesService {
	
    // Insert.
	Result insert(final DeliveryModes deliveryModes) ;

    // Update.
	Result update(final DeliveryModes deliveryModes) ;

    // Delete.
	Result delete(final Integer id) ;

	// Get By Id.
	Result getById(final Integer id) ;

    // Exists.
	Result exists(final Integer id) ;

    // Count.
	Result count() ;

	// List for page and filter.
	Result listDeliveryModesForPageAndFilter(final int itemsPerPage, final int pageNo, final DeliveryModes deliveryModes);

}
